primefaces/src/main/resources/META-INF/resources/primefaces/forms/forms.selectcheckboxmenu.js

```diff
@@ -651,7 +656,21 @@ PrimeFaces.widget.SelectCheckboxMenu = PrimeFaces.widget.BaseWidget.extend({
 
         var itemKeyInputs = this.itemContainer.find('> li > div.ui-chkbox > div.ui-helper-hidden-accessible > input');
         this.bindCheckboxKeyEvents(itemKeyInputs);
-        itemKeyInputs.on('keyup.selectCheckboxMenu', function(e) {
+        itemKeyInputs.on('keydown.selectCheckboxMenu', function(e) {
+            var input = $(this),
+                item = input.closest("li"),
+                index = $this.items.index(item);
+
+            if(e.which === $.ui.keyCode.TAB && !e.shiftKey
+                && index === $this.items.length - 1) {
+                e.preventDefault();
+            }
+            else if(e.which === $.ui.keyCode.TAB && e.shiftKey
+                && !$this.cfg.showHeader && index === 0) {
+                e.preventDefault();
+            }
```

Comment:
Slightly more readable might be...
```js
if (e.which === $.ui.keyCode.TAB) {
    if (!e.shiftKey && index === $this.items.length - 1) {
        e.preventDefault();
    } else if (e.shiftKey && !$this.cfg.showHeader && index === 0) {
        e.preventDefault();
    }
}
```

Reply:
good point, has been added in my new commit